src/pages/task/ActionButtons.tsx

```diff
@@ -0,0 +1,247 @@
+      success("Task marked as unscheduled");
+    },
+    onError: (err) => {
+      error(`Error unscheduling task: ${err.message}`);
+    },
+    refetchQueries: ["GetTask"],
+  });
+
+  const [abortTask, { loading: loadingAbortTask }] = useMutation<
+    AbortTaskMutation,
+    AbortTaskMutationVariables
+  >(ABORT_TASK, {
+    variables: {
+      taskId,
+    },
+    onCompleted: () => {
+      success("Task aborted");
+    },
+    onError: (err) => {
+      error(`Error aborting task: ${err.message}`);
+    },
+    refetchQueries: ["GetTask"],
+  });
+
+  const [restartTask, { loading: loadingRestartTask }] = useMutation<
+    RestartTaskMutation,
+    RestartTaskMutationVariables
+  >(RESTART_TASK, {
+    variables: { taskId },
+    onCompleted: () => {
+      success("Task scheduled to restart");
+    },
+    onError: (err) => {
+      error(`Error restarting task: ${err.message}`);
+    },
+    refetchQueries: ["GetTask"],
+  });
+
+  const [setTaskPriority, { loading: loadingSetPriority }] = useMutation<
+    SetTaskPriorityMutation,
+    SetTaskPriorityMutationVariables
+  >(SET_TASK_PRIORTY, {
+    onCompleted: (data) => {
+      success(`Priority for task updated to ${data.setTaskPriority.priority}`);
+    },
+    onError: (err) => {
+      error(`Error updating priority for task: ${err.message}`);
+    },
+    refetchQueries: ["GetTask"],
+  });
+
+  const disabled =
+    loadingAbortTask ||
+    loadingRestartTask ||
+    loadingSetPriority ||
+    loadingUnscheduleTask ||
+    loadingScheduleTask;
+
+  useEffect(() => {
+    if (disabled) {
+      setIsVisible(false);
+    }
+  }, [disabled]);
+
+  useOnClickOutside(wrapperRef, () => {
+    if (
+      !get(priorityRef, "current.className", "").includes("ant-popover-open")
+    ) {
+      setIsVisible(false);
+    }
+  });
+
+  const rowButtons = [
+    <Button
+      size="small"
+      dataCy="schedule-task"
+      key="schedule"
+      disabled={disabled || !canSchedule}
+      loading={loadingScheduleTask}
+      onClick={scheduleTask}
+    >
+      Schedule
+    </Button>,
+    <Button
+      size="small"
+      dataCy="restart-task"
+      key="restart"
+      disabled={disabled || !canRestart}
+      loading={loadingRestartTask}
+      onClick={restartTask}
+    >
+      Restart
+    </Button>,
+    <Button
+      size="small"
+      dataCy="notify-task"
+      key="notifications"
+      disabled={disabled}
+    >
+      Add Notification
+    </Button>,
+  ];
+
+  const cardItems = [
+    <CardItem
+      disabled={disabled || !canUnschedule}
+      key="unschedule"
+      data-cy="unschedule-task"
+      onClick={() => unscheduleTask()}
+    >
+      <Body>Unschedule</Body>
+    </CardItem>,
+    <CardItem
+      data-cy="abort-task"
+      key="abort"
+      disabled={disabled || !canAbort}
+      onClick={() => abortTask()}
+    >
+      <Body>Abort</Body>
+    </CardItem>,
+    <Popconfirm
+      key="priority"
+      placement="left"
+      title={
```

Comment:
looks like the default icon for this component is a warning sign - can we pass null to suppress it?

Reply:
@SupaJoon did you confirm with lara/design that using the popconfirm is goodie?